tests/integration/test_cross_model_remote_write.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import asyncio
+import logging
+
+import pytest
+from helpers import check_prometheus_is_ready, oci_image, run_promql
+from juju.controller import Controller  # type: ignore
+
+logger = logging.getLogger(__name__)
+
+prometheus_name = "prometheus"
+agent_name = "grafana-agent"
+
+
+@pytest.mark.abort_on_fail
+async def test_create_remote_write_models(ops_test, prometheus_charm):
+    """Test that Prometheus can be related with the Grafana Agent over remote_write."""
+    await ops_test.model.deploy(
```

Comment:
Move this deploy below and into a `gather`?
```suggestion
    await offer.model.deploy(
```